Add support for oneCore, so MavLinkCom works on Raspberry Pi.

lovettchris opened the pull request.

Commit:
Add support for oneCore, so MavLinkCom works on Raspberry Pi.

msftclas commented:
@lovettchris,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

andre-nguyen commented on .gitignore:
```diff
@@ -298,7 +298,7 @@ deps/
 
 /boost
-/eigen
+/Ee]igen
```
Typo?

lovettchris replied:
Fixed, thanks.

Commit:
fix typo

sytelus commented:
Couple of asks...

1. Can we create separate MavLinkCom.vsxproj for VS2017? For example, MavLinkCom_VS2017.vcxproj. All our other project are in VS2015. We can then have MavLinkCom_VS2017.sln.

2. I'm getting following warning. Not sure what it means:
```
1>  FileSystem.cpp
1>  Generating Code...
1>OneCoreFindSerialPorts.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that consumes this library
1>  MavLinkCom.vcxproj -> C:\GitHubSrc\AirSim\MavLinkCom\lib\x64\Debug\MavLinkCom.lib
2>------ Rebuild All started: Project: AirLib, Configuration: Debug x64 ------
3>------ Rebuild All started: Project: MavLinkTest, Configuration: Debug x64 ------

```

3. It looks like MavLinkTest depenencies is changed from 
`<AdditionalDependencies>kernel32.lib;user32.lib;uuid.lib;%(AdditionalDependencies)</AdditionalDependencies>`

To.
`      <AdditionalDependencies>onecore.lib;;%(AdditionalDependencies)</AdditionalDependencies>`

I'm not familier with onecore too much but would this be compilable on at least all Win 7 and Win 10 systems? If not, may be we should create different cxproj for onecore?

4. I'd made few changes to MavLinkFindSerialPorts some time ago however as this file is renamed, so just wanted to confirm if your changes are on top of those changes...

Commit:
Merge branch 'master' into review/onecore

Commit:
Fix review feedback.

lovettchris commented:
I have updated the *.vcxproj files so they work on VS 2015 and I have modified it so that onecore.lib is only used for ARM configuration, leaving the win32,win64 route the same as it was before (so it still supports windows 8.1 - which was the old target version by the way).  I tested that these bits work for big windows 10 and for windows IOT on raspberry pi.

sytelus commented:
Thanks! It looks great. I'm getting this warning when compiling from scratch:

```
"C:\GitHubSrc\AirSim\AirSim.sln" (default target) (1) ->
"C:\GitHubSrc\AirSim\DroneShell\DroneShell.vcxproj" (default target) (2) ->
"C:\GitHubSrc\AirSim\AirLib\AirLib.vcxproj" (default target) (3) ->
"C:\GitHubSrc\AirSim\MavLinkCom\MavLinkCom.vcxproj" (default target) (4) ->
(Lib target) ->
  OneCoreFindSerialPorts.obj : warning LNK4221: This object file does not define any previously undefined public symbol
s, so it will not be used by any link operation that consumes this library [C:\GitHubSrc\AirSim\MavLinkCom\MavLinkCom.v
cxproj]

    1 Warning(s)
    0 Error(s)

```

Here are some ways to deal with it: https://stackoverflow.com/questions/20854324/how-to-eliminate-warning-lnk4221

Please let me know if we can get rid of this warning...

sytelus commented:
Also, is VS 2017 upgrade required for MavLinkCom.sln? I'd added find serial port code for Linux as well and wondering if that would still work...

Commit:
Merge branch 'master' into review/onecore

Commit:
revert unnecessary VS version change.

lovettchris commented:
I reverted the change in VS version in MavLinkCom.sln and verified the onecore ARM build of MavLinkCom works from VS 2015.

sytelus commented:
Not sure what's wrong but when I get this branch, I get huge change set. I think it might be cause of some merge that happened at your end and may be your branch was not rebased? Anyway, I'd like to get this changeset and looks like there are 11 files affected. It would be great to get one clean PR that has only these files.

```
[C:\GitHubSrc\AirSim>git stat
On branch review/onecore
Your branch is up-to-date with 'origin/review/onecore'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   AirLib/AirLib.vcxproj
        modified:   AirLib/AirLib.vcxproj.filters
        modified:   AirLib/include/api/RpcLibAdapatorsBase.hpp
        copied:     AirLib/include/vehicles/car/api/CarRpcLibClient.hpp -> AirLib/include/api/RpcLibClientBase.hpp
        modified:   AirLib/include/common/VectorMath.hpp
        modified:   AirLib/include/common/common_utils/FileSystem.hpp
        modified:   AirLib/include/common/common_utils/Utils.hpp
        modified:   AirLib/include/controllers/VehicleCameraBase.hpp
        renamed:    AirLib/include/vehicles/car/controllers/CarControllerBase.hpp -> AirLib/include/vehicles/car/api/CarApiBase.hpp
        modified:   AirLib/include/vehicles/car/api/CarRpcLibAdapators.hpp
        modified:   AirLib/include/vehicles/car/api/CarRpcLibClient.hpp
        modified:   AirLib/include/vehicles/car/api/CarRpcLibServer.hpp
        modified:   AirLib/include/vehicles/multirotor/api/MultirotorRpcLibClient.hpp
        modified:   AirLib/include/vehicles/multirotor/controllers/DroneControllerBase.hpp
        modified:   AirLib/include/vehicles/multirotor/controllers/DroneControllerCancelable.hpp
        modified:   AirLib/include/vehicles/multirotor/firmwares/simple_flight/SimpleFlightDroneController.hpp
        copied:     AirLib/src/vehicles/car/api/CarRpcLibClient.cpp -> AirLib/src/api/RpcLibClientBase.cpp
        modified:   AirLib/src/common/common_utils/FileSystem.cpp
        modified:   AirLib/src/controllers/DroneControllerBase.cpp
        modified:   AirLib/src/vehicles/car/api/CarRpcLibClient.cpp
        modified:   AirLib/src/vehicles/car/api/CarRpcLibServer.cpp
        modified:   AirLib/src/vehicles/multirotor/api/MultirotorRpcLibClient.cpp
        modified:   AirLib/src/vehicles/multirotor/api/MultirotorRpcLibServer.cpp
        modified:   AirSim.sln
        modified:   DroneShell/src/main.cpp
        modified:   Examples/StereoImageGenerator.hpp
        modified:   HelloCar/main.cpp
        modified:   HelloDrone/main.cpp
        renamed:    PythonClient/PythonClient.py -> PythonClient/AirSimClient.py
        modified:   PythonClient/PythonClient.pyproj
        modified:   PythonClient/box.py
        modified:   PythonClient/camera.py
        modified:   PythonClient/cv_mode.py
        modified:   PythonClient/hello_car.py
        modified:   PythonClient/hello_drone.py
        modified:   PythonClient/navigate.py
        modified:   PythonClient/path.py
        modified:   PythonClient/point_cloud.py
        modified:   README.md
        modified:   Unreal/Environments/Blocks/Blocks.uproject
        modified:   Unreal/Environments/Blocks/clean.bat
        modified:   Unreal/Plugins/AirSim/AirSim.uplugin
        modified:   Unreal/Plugins/AirSim/Content/Blueprints/BP_FlyingPawn.uasset
        modified:   Unreal/Plugins/AirSim/Content/Blueprints/BP_PIPCamera.uasset
        modified:   Unreal/Plugins/AirSim/Content/Blueprints/BP_SimHUDWidget.uasset
        modified:   Unreal/Plugins/AirSim/Content/HUDAssets/ColorizeMatarialFunction.uasset
        modified:   Unreal/Plugins/AirSim/Content/HUDAssets/DepthMaterialFunction.uasset
        deleted:    Unreal/Plugins/AirSim/Content/HUDAssets/DepthMetersMaterial.uasset
        new file:   Unreal/Plugins/AirSim/Content/HUDAssets/DepthPerspectiveMaterial.uasset
        new file:   Unreal/Plugins/AirSim/Content/HUDAssets/DepthPlannerMaterial.uasset
        modified:   Unreal/Plugins/AirSim/Content/HUDAssets/DepthVisMaterial.uasset
        modified:   Unreal/Plugins/AirSim/Content/HUDAssets/DisparityMaterial.uasset
        modified:   Unreal/Plugins/AirSim/Source/AirBlueprintLib.cpp
        modified:   Unreal/Plugins/AirSim/Source/AirBlueprintLib.h
        modified:   Unreal/Plugins/AirSim/Source/AirSimGameMode.cpp
        modified:   Unreal/Plugins/AirSim/Source/AirSimGameMode.h
        modified:   Unreal/Plugins/AirSim/Source/CameraDirector.cpp
        modified:   Unreal/Plugins/AirSim/Source/Car/CarPawn.cpp
        modified:   Unreal/Plugins/AirSim/Source/Car/CarPawn.h
        modified:   Unreal/Plugins/AirSim/Source/Car/SimModeCar.cpp
        modified:   Unreal/Plugins/AirSim/Source/Car/SimModeCar.h
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/FlyingPawn.cpp
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/FlyingPawn.h
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/MultiRotorConnector.cpp
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/MultiRotorConnector.h
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/SimModeWorldMultiRotor.cpp
        modified:   Unreal/Plugins/AirSim/Source/Multirotor/SimModeWorldMultiRotor.h
        modified:   Unreal/Plugins/AirSim/Source/PIPCamera.cpp
        modified:   Unreal/Plugins/AirSim/Source/PIPCamera.h
        modified:   Unreal/Plugins/AirSim/Source/Recording/RecordingFile.cpp
        modified:   Unreal/Plugins/AirSim/Source/Recording/RecordingFile.h
        new file:   Unreal/Plugins/AirSim/Source/Recording/RecordingSettings.h
        modified:   Unreal/Plugins/AirSim/Source/Recording/RecordingThread.cpp
        modified:   Unreal/Plugins/AirSim/Source/Recording/RecordingThread.h
        modified:   Unreal/Plugins/AirSim/Source/SimHUD/SimHUD.cpp
        modified:   Unreal/Plugins/AirSim/Source/SimMode/SimModeBase.cpp
        modified:   Unreal/Plugins/AirSim/Source/SimMode/SimModeBase.h
        modified:   Unreal/Plugins/AirSim/Source/SimMode/SimModeWorldBase.cpp
        modified:   Unreal/Plugins/AirSim/Source/SimMode/SimModeWorldBase.h
        modified:   Unreal/Plugins/AirSim/Source/VehiclePawnWrapper.cpp
        modified:   Unreal/Plugins/AirSim/Source/VehiclePawnWrapper.h
        new file:   UnrealPluginFiles.vcxproj
        new file:   UnrealPluginFiles.vcxproj.filters
        modified:   cmake/AirLib/CMakeLists.txt
        modified:   docs/apis.md
        modified:   docs/image_apis.md
        new file:   docs/pfm.md
        modified:   docs/px4_setup.md
        modified:   docs/python.md
        modified:   docs/settings.md
        modified:   docs/unreal_blocks.md
        modified:   docs/using_car.md
        modified:   external/rpclib](url)
```

Commit:
Merge branch 'master' into review/onecore

clovett commented:
Perhaps it was just out of date relative to master.  I just merged master into this branch, so it should now be up to date, can you try again?

sytelus commented:
Thanks. I've merged this PR: https://github.com/Microsoft/AirSim/commit/cc7590694c8e9bda39bda28a9e8637098c45c8db
